cmd/system/containerd.go

```diff
@@ -0,0 +1,194 @@
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		installPath, _ := cmd.Flags().GetString("path")
+		version, _ := cmd.Flags().GetString("version")
+		progress, _ := cmd.Flags().GetBool("progress")
+		systemd, _ := cmd.Flags().GetBool("systemd")
+		archFlag, _ := cmd.Flags().GetString("arch")
+
+		arch, osVer := env.GetClientArch()
+		if archFlag != "" {
+			arch = archFlag
+		}
+
+		fmt.Printf("Installing containerd to %s\n", installPath)
+
+		if strings.ToLower(osVer) != "linux" {
+			return fmt.Errorf("this app currently only supports Linux")
+		}
+
+		if version == "" {
+			latestVerison, err := get.FindGitHubRelease("containerd", "containerd")
+			if err != nil {
+				return err
+			}
+			version = latestVerison
+		}
+
+		downloadArch := ""
+
+		if arch == "x86_64" {
+			downloadArch = "amd64"
+		} else if arch == "aarch64" {
+			downloadArch = "arm64"
+		} else {
+			return fmt.Errorf("this app currently only supports arm64 and amd64 archs")
+		}
+
+		containerdTool := get.Tool{
+			Name:    "containerd",
+			Repo:    "containerd",
+			Owner:   "containerd",
+			Version: version,
+			BinaryTemplate: `
+			{{$archStr := .Arch}}
+			{{- if eq .Arch "aarch64" -}}
+			{{$archStr = "arm64"}}
+			{{- else if eq .Arch "x86_64" -}}
+			{{$archStr = "amd64"}}
+			{{- end -}}
+			{{.Name}}-{{.VersionNumber}}-{{.OS}}-{{$archStr}}.tar.gz
+			`,
+		}
+
+		url, err := containerdTool.GetURL(osVer, downloadArch, containerdTool.Version, !progress)
+		if err != nil {
+			return err
+		}
+
+		outPath, err := get.DownloadFileP(url, progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloaded to: %s\n", outPath)
+
+		f, err := os.OpenFile(outPath, os.O_RDONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		tempDirName := os.TempDir() + "/containerd"
+
+		if err := archive.UntarNested(f, tempDirName); err != nil {
+			return err
+		}
+
+		fmt.Printf("Copying containerd binaries to: %s\n", installPath)
+
+		dir, err := os.ReadDir(tempDirName + "/bin")
+		if err != nil {
+			return err
+		}
+
+		if err := os.MkdirAll(installPath, 0755); err != nil && !os.IsExist(err) {
+			fmt.Printf("Error creating directory %s, error: %s\n", installPath, err.Error())
+		}
+
+		for _, entry := range dir {
+			_, err := get.CopyFile(tempDirName+"/bin/"+entry.Name(), installPath+"/"+entry.Name())
+			if err != nil {
+				return err
+			}
+		}
+
+		if systemd {
+			systemdUnitName := "containerd.service"
+			systemdUnitUrl := fmt.Sprintf("https://raw.githubusercontent.com/containerd/containerd/%s/%s", version, systemdUnitName)
+
+			response, err := http.Get(systemdUnitUrl)
+			if err != nil {
+				return err
+			}
+
+			content, err := ioutil.ReadAll(response.Body)
+			if err != nil {
+				return err
+			}
+
+			content = bytes.ReplaceAll(content, []byte("/usr/local/bin/containerd"), []byte(installPath+"/containerd"))
+			content = bytes.Join([][]byte{content, []byte("[Manager]"), []byte("DefaultTimeoutStartSec=3m")}, []byte("\n"))
```

Comment:
Add a comment as to why this is done for future maintainers. See the faasd hack/install.sh for the reason